internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -140,57 +142,63 @@ func resourceCdnFrontdoorSecurityPolicyCreate(d *pluginsdk.ResourceData, meta in
 	defer cancel()
 
 	// NOTE: The profile id is used to retrieve properties from the related profile that must match in this security policy
-	profile, err := parse.FrontDoorProfileID(d.Get("cdn_frontdoor_profile_id").(string))
+	profileId, err := profiles.ParseProfileID(d.Get("cdn_frontdoor_profile_id").(string))
 	if err != nil {
 		return err
 	}
 
 	securityPolicyName := d.Get("name").(string)
-	id := parse.NewFrontDoorSecurityPolicyID(profile.SubscriptionId, profile.ResourceGroup, profile.ProfileName, securityPolicyName)
+	id := securitypolicies.NewSecurityPolicyID(profileId.SubscriptionId, profileId.ResourceGroupName, profileId.ProfileName, securityPolicyName)
 
-	existing, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.SecurityPolicyName)
+	existing, err := client.Get(ctx, id)
 	if err != nil {
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return fmt.Errorf("checking for existing %s: %+v", id, err)
 		}
 	}
 
-	if !utils.ResponseWasNotFound(existing.Response) {
+	if !response.WasNotFound(existing.HttpResponse) {
 		return tf.ImportAsExistsError("azurerm_cdn_frontdoor_security_policy", id.ID())
 	}
 
-	profileClient := meta.(*clients.Client).Cdn.FrontDoorProfileClient
-	resp, err := profileClient.Get(ctx, profile.ResourceGroup, profile.ProfileName)
+	profileClient := meta.(*clients.Client).Cdn.FrontDoorProfilesClient
+	resp, err := profileClient.Get(ctx, pointer.From(profileId))
 	if err != nil {
-		return fmt.Errorf("unable to retrieve the 'sku_name' from the CDN FrontDoor Profile(Name: %q)': %+v", profile.ProfileName, err)
+		return fmt.Errorf("unable to retrieve the 'sku_name' from the CDN FrontDoor Profile(Name: %q)': %+v", profileId.ProfileName, err)
 	}
 
-	if resp.Sku == nil {
-		return fmt.Errorf("the CDN FrontDoor Profile(Name: %q) 'sku' was nil", profile.ProfileName)
+	profileModel := resp.Model
+
+	if profileModel == nil {
+		return fmt.Errorf("profileModel is 'nil'")
+	}
+
+	if profileModel.Properties == nil {
+		return fmt.Errorf("profileModel.Properties is 'nil'")
+	}
+
+	if profileModel.Sku.Name == nil {
+		return fmt.Errorf("profileModel.Sku.Name is 'nil'")
 	}
```

Comment:
Do we need this since we're checking the value of this safely below in for the `isStandardSku` bool below?
```suggestion
```

Reply:
I believe we should still check for `nil` value else it will panic?